### PR TITLE
managesieve: init at 0.6

### PR DESCRIPTION
**Pull request by dadada**
###### Motivation for this change

Add managesieve python library and sieveshell script.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @htgoebel

### CODE REVIEW

File: pkgs/development/python-modules/managesieve/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "ee70e298e9b68eb81f93d52a1320a034fdc182f3927fdd551836fc93b0ed2c5f";
+  };
+
+  doCheck = true;
+
+  checkInputs = [ pytestrunner pytest ];
+  propagatedBuildInputs = [ ];
```
Comment (teto) [inline pkgs/development/python-modules/managesieve/default.nix]: you can remove this as well as the doCheck which defaults to true;

File: pkgs/development/python-modules/managesieve/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description = "ManageSieve client library for remotely managing Sieve scripts";
+    homepage    = "https://managesieve.readthedocs.io/";
+    license     = [ licenses.gpl3 licenses.psfl ];
```
Comment (teto) [inline pkgs/development/python-modules/managesieve/default.nix]: maybe add a comment as to why there are 2 licenses:
Python Software Foundation License for the module, GPL v3 for sieveshell and test suite.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -25148,4 +25148,6 @@ in
 
   keycard-cli = callPackage ../tools/security/keycard-cli {};
 
+  managesieve = python3Packages.callPackage ../development/python-modules/managesieve {};
```
Comment (teto) [inline pkgs/top-level/all-packages.nix]: if it's a package it should go python-packages.nix. if it's an application you should use buildPythonPackage.
if it's because siveshell is an application, you can build it as package in python-packages.nix and alias sieveshell = toPythonApplication (managesieve), you can find examples in nixpkgs.

Comment (dadada) [reply pkgs/top-level/all-packages.nix]: Thanks for the helpful advice

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -1415,6 +1415,16 @@
...
+  dadada = {
+    name = "dadada";
+    email = "dadada@dadada.li";
+    github = "dadada";
+    githubId = 7216772;
+    keys = [{
+      longkeyid = "ed25519/0xEEB8D1CE62C4DFEA";
+      fingerprint = "D68C 8469 5C08 7E0F 733A  28D0 EEB8 D1CE 62C4 DFEA";
+    }];
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: this should be a separate commit:
```
maintainers: add dadada
```

File: pkgs/development/python-modules/managesieve/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchPypi
+, pytestrunner
+, pytest
+}:
+
+with stdenv.lib;
```
Comment (jonringer) [inline pkgs/development/python-modules/managesieve/default.nix]: unused
```suggestion
```

File: pkgs/development/python-modules/managesieve/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{ lib
+, stdenv
```
Comment (jonringer) [inline pkgs/development/python-modules/managesieve/default.nix]: unused
```suggestion
```

File: pkgs/development/python-modules/managesieve/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "ee70e298e9b68eb81f93d52a1320a034fdc182f3927fdd551836fc93b0ed2c5f";
+  };
+
+  checkInputs = [ pytestrunner pytest ];
+
+  meta = with lib; {
+    description = "ManageSieve client library for remotely managing Sieve scripts";
+    homepage    = "https://managesieve.readthedocs.io/";
+    # PSFL for the python module, GPLv3 for sieveshell
+    license     = [ licenses.gpl3 licenses.psfl ];
```
Comment (jonringer) [inline pkgs/development/python-modules/managesieve/default.nix]:
```suggestion
    license     = with licenses; [ gpl3 psfl ];
```

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS (python38 broken due to dependency broken (pytest))
diff LGTM
executable seems to work
leaf package

```
[3 built (1 failed), 0.0 MiB DL]
error: build of '/nix/store/ydvfxh1lvfyjr73z03a4nzads55sv0c3-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/71028
1 package failed to build:
python38Packages.managesieve

2 package were build:
python27Packages.managesieve sieveshell
```
```
[nix-shell:/home/jon/.cache/nix-review/pr-71028]$ nix path-info -Sh ./results/python27Packages.managesieve
/nix/store/jdq5hlyfj8z36m48akagx1r3m802bzcz-python2.7-managesieve-0.6     93.7M
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.managesieve sieveshell

**Comment by jonringer** [comment]
oops, looks like I forgot to push the button, thanks for staying through all the revisions :)

**Comment by htgoebel** [comment]
Congratulations and Thanks. AFAIK NixOs is the first distribution to integrate my managesieve.
